wgpu/utils/imgui_backend.py

```diff
@@ -0,0 +1,510 @@
+            # )
+            # index_mem_address = commands.idx_buffer.data_address()
+            # self._device.queue.write_buffer(
+            #     self._index_buffer, global_idx_offset* imgui.INDEX_SIZE, ctypes.string_at(ctypes.c_void_p(index_mem_address), commands.idx_buffer.size() * imgui.INDEX_SIZE)
+            # )
+
+            for command in commands.cmd_buffer:
+                # todo command.user_callback
+
+                tex_id = command.texture_id
+
+                if tex_id not in self._image_bind_groups:
+                    image_bind_group = self._device.create_bind_group(
+                        layout=self.image_bind_group_layout,
+                        entries=[
+                            {
+                                "binding": 0,
+                                "resource": self._texture_views[tex_id],
+                            }
+                        ],
+                    )
+                    self._image_bind_groups[tex_id] = image_bind_group
+
+                render_pass.set_bind_group(1, self._image_bind_groups[tex_id])
+
+                clip_rect = command.clip_rect
+                clip_min = [
+                    (clip_rect.x - clip_off.x) * clip_scale.x,
+                    (clip_rect.y - clip_off.y) * clip_scale.y,
+                ]
+                clip_max = [
+                    (clip_rect.z - clip_off.x) * clip_scale.x,
+                    (clip_rect.w - clip_off.y) * clip_scale.y,
+                ]
+                if clip_min[0] < 0:
+                    clip_min[0] = 0
+                if clip_min[1] < 0:
+                    clip_min[1] = 0
+                if clip_max[0] > fb_width:
+                    clip_max[0] = fb_width
+                if clip_max[1] > fb_height:
+                    clip_max[1] = fb_height
+
+                if clip_max[0] - clip_min[0] <= 0 or clip_max[1] - clip_min[1] <= 0:
+                    continue
+
+                render_pass.set_scissor_rect(
+                    int(clip_min[0]),
+                    int(clip_min[1]),
+                    int(clip_max[0] - clip_min[0]),
+                    int(clip_max[1] - clip_min[1]),
+                )
+
+                render_pass.draw_indexed(
+                    command.elem_count,
+                    1,
+                    command.idx_offset + global_idx_offset,
+                    command.vtx_offset + global_vtx_offset,
+                    0,
+                )
+
+            global_vtx_offset += commands.vtx_buffer.size()
+            global_idx_offset += commands.idx_buffer.size()
+
+    def _invalidate_device_objects(self):
+        self._render_pipeline._destroy()
+        self._uniform_buffer._destroy()
+        self._font_texture._destroy()
+        self._font_texture_view._destroy()
+        self._font_texture_sampler._destroy()
```

Comment:
Using private methods here. These were also changed recently. Can `.destroy()` textures and buffers. The rest can simply be set to None, I think, to let the gc do its thing.

Reply:
I recently tested and discovered that after the update, actively calling the destroy method on an existing texture while recreating the font texture in the `create_fonts_texture` method causes a Rust panic (texture already destroyed). It seems the old font texture cannot be immediately destroyed here. Therefore, I removed the explicit destroy call and will let the garbage collector handle it instead.

Reply:
To provide a bit of context: the previous destroy methods in wgpu-py actually called `xxRelease`, which basically reduces the ref-counter for that object, i.e. Rust knows that we no longer use it. The new `destroy()` methods in wgpu-py call the actual destroy functions in wgpu-native, so the low-level object is marked as not-usable, but this can indeed lead to errors when the object is still in use internally.